### ISSUE TITLE
Shouldn't TransferState.set be called only on server side?

### ISSUE THREAD

**Issue by zip-fa** [angular/angular#54444]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Hi.

I noticed that this piece of code in [packages/common/http/src/transfer_cache.ts](https://github.com/angular/angular/blob/9ca8c6852896a21d8edfeb78dfb99a500e2dd281/packages/common/http/src/transfer_cache.ts#L198) gets called on client side:

```ts
return next(req).pipe(
    tap((event: HttpEvent<unknown>) => {
      if (event instanceof HttpResponse) {
        transferState.set<TransferHttpResponse>(storeKey, {
          [BODY]: event.body,
          [HEADERS]: getFilteredHeaders(event.headers, headersToInclude),
          [STATUS]: event.status,
          [STATUS_TEXT]: event.statusText,
          [URL]: event.url || '',
          [RESPONSE_TYPE]: req.responseType,
        });
      }
    }),
  );
```

Shouldn't this only be called only on server side? It wastes some memory on client's device

**Comment by alan-agius4** [comment]
Technically this could only be called on the client side when;
 - The cache is in active, meaning that application has not been stabilised yet.
 - The request was not already performed on the server. 
 
 In general, I'd say that this is rather quite uncommon, although I don't have anything against adding the additional check.

**Comment by dchicurel** [comment]
Adding +1 as we just found this issue. For some reasons, we needed to have the app deployed either as CSR as SSR and found this issue. Transfer state seems to be working just as a store in client mode, so it's caching the current actual client requests.

**Comment by dchicurel** [comment]
We just found that transfer cache should only work during App bootstrapping and it's being disabled after the app is stable. That was well thought. I didn't noticed in your comment @alan-agius4 when you mentioned about the app not "stabilised" 😊 

https://github.com/angular/angular/blob/5bd188a394d30053099e2c83fe79136d590e5399/packages/common/http/src/transfer_cache.ts#L264-L274

Our problem was that, actually, the App was not being Stable! We fixed that and confirmed that cache is disabled after the app becomes stable (first time).
Now it makes sense your comment also that this (original) issue would be an uncommon case.
